# IS 362 – Project 1
## Analysis of Airline Arrival Delays

This project compares arrival delays for Alaska Airlines and AM West across five destinations: Los Angeles, Phoenix, San Diego, San Francisco, and Seattle.

The data is stored in a CSV file and will be analyzed using pandas. Because the airlines have different numbers of flights, I will compare their delay percentages rather than relying only on the number of delayed flights.

In [4]:
import pandas as pd

airlines = pd.read_csv("/Users/alyssaellis/Desktop/airline - Sheet1.csv")
airlines

,Airline,Status,Los Angeles,Phoenix,San Diego,San Francisco,Seattle
0,Alaska,On Time,497,221,212,503,1841
1,Alaska,Delayed,62,12,20,102,305
2,AM West,On Time,694,4840,383,320,201
3,AM West,Delayed,117,415,65,129,61


In [5]:
# Separate the on-time and delayed flights
on_time = airlines[airlines["Status"] == "On Time"].set_index("Airline")
delayed = airlines[airlines["Status"] == "Delayed"].set_index("Airline")

# Remove the Status column
on_time = on_time.drop(columns="Status")
delayed = delayed.drop(columns="Status")

# Calculate the total number of flights
total_flights = on_time + delayed

# Calculate the delay percentage for each destination
delay_percentage = (delayed / total_flights) * 100

delay_percentage.round(2)

,Los Angeles,Phoenix,San Diego,San Francisco,Seattle
Airline,,,,,
Alaska,11.09,5.15,8.62,16.86,14.21
AM West,14.43,7.90,14.51,28.73,23.28


In [6]:
# Calculate total delayed flights for each airline
total_delayed = delayed.sum(axis=1)

# Calculate total flights for each airline
total_all_flights = total_flights.sum(axis=1)

# Calculate overall delay percentage
overall_delay_percentage = (total_delayed / total_all_flights) * 100

overall_delay_percentage.round(2)

Airline
Alaska     13.27
AM West    10.89
dtype: float64

## Analysis and Conclusion

The results show that Alaska had a lower percentage of delayed flights than AM West at each of the five destinations.

For Los Angeles, Alaska's delay rate was 11.09%, compared with 14.43% for AM West. In Phoenix, Alaska had a delay rate of 5.15%, compared with 7.90% for AM West. In San Diego, Alaska's delay rate was 8.62%, while AM West's was 14.51%. In San Francisco, Alaska had a delay rate of 16.86%, compared with 28.73% for AM West. In Seattle, Alaska's delay rate was 14.21%, while AM West's was 23.28%.

However, when all five destinations are combined, Alaska's overall delay rate is approximately 13.27%, while AM West's overall delay rate is approximately 10.89%.

This happens because the two airlines have very different numbers of flights going to each destination. AM West has a particularly large number of flights to Phoenix, where its delay rate is relatively low. Therefore, the overall percentages give a different result from the destination-by-destination comparison.

Based on the destination-level comparison, Alaska had a lower delay percentage at every destination in this dataset. This analysis demonstrates why percentages and the distribution of flights should be considered when comparing airline performance.